src/client.cpp

```diff
@@ -819,22 +819,14 @@ static int _connect_check_vma(int ifd) {
 //------------------------------------------------------------------------------
 static int _connect_check(int ifd) {
     int rc = SOCKPERF_ERR_NONE;
-    fd_set rfds, wfds;
-    struct timeval tv;
-    tv.tv_sec = 0;
-    tv.tv_usec = 500;
-
-    FD_ZERO(&rfds);
-    FD_ZERO(&wfds);
-
-    int max_fd = -1;
-
-    FD_SET(ifd, &wfds);
-    FD_SET(ifd, &rfds);
-    if (ifd > max_fd) max_fd = ifd;
-
-    select(max_fd + 1, &rfds, &wfds, NULL, &tv);
-    if (FD_ISSET(ifd, &wfds) || FD_ISSET(ifd, &rfds)) {
+    struct pollfd fds = { .fd = ifd, .events = POLLIN | POLLOUT, };
+#define POLL_TIMEOUT_MS 1
+    int ret = poll(&fds, 1, POLL_TIMEOUT_MS);
```

Comment:
please do it under `#ifndef WIN32` and keep `select()` approach as temporary solution for windows

Reply:
done.